flake.nix

```diff
@@ -0,0 +1,48 @@
+{
+  description = "A Nix-flake-based development environment for Cursorless";
+
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+  outputs = { self, nixpkgs }:
+    let
+      overlays = [
+        (final: prev: rec {
+          nodejs = prev.nodejs-18_x;
+          pnpm = prev.nodePackages.pnpm;
```

Comment:
Should we use corepack instead? Odds are this will break when someone updates the pnpm version in package.json without updating the flake.lock.

Breaking changes in pnpm aren't unprecedented either; pnpm@9 completely changed the lockfile format.

Reply:
First I've heard of it, but certainly sounds/looks reasonable. I'll test it out tomorrow and if I don't run into any bumps I'll switch it over.

Reply:
Seems to work fine, so removed the overlay. Also reformatted using nixftm-rfc-style since it's the new nixpkgs standard.